### PR TITLE
Fix deterministic behavior in `move_agent_to_one_of` with `selection="closest"`

### PR DESCRIPTION
**Pull request by OrenBochman**
This PR addresses a bug in `move_agent_to_one_of` where agents exhibit deterministic behavior when multiple closest positions are available and `selection="closest"` is used. This issue caused agents to follow periodic motion patterns instead of random selection within sugar plateaus, particularly noticeable in low-vision agents.

**Changes**:
- Modified `move_agent_to_one_of` to ensure randomness when multiple equally distant positions are found by shuffling the positions and randomly selecting from the closest ones.
- Added a new test, `test_move_agent_closest_selection_multiple`, to verify that agents exhibit random behavior in scenarios with multiple closest positions.

**Fixes**: Agents now select randomly from multiple closest positions, preventing deterministic motion.

Closes: https://github.com/projectmesa/mesa/issues/2117

### CODE REVIEW

File: mesa/space.py [hunk 1]
```diff
@@ -459,14 +459,19 @@ def move_agent_to_one_of(
             elif selection == "closest":
                 current_pos = agent.pos
                 # Find the closest position without sorting all positions
-                closest_pos = None
+                closest_pos = []
                 min_distance = float("inf")
                 for p in pos:
```
Comment (EwoutH) [inline mesa/space.py:464]: What if instead of keeping a list, we shuffle the `pos` list here? I think that's both faster and cleaner in implementation.

**Review by EwoutH — COMMENTED** [review]
Thanks a lot for catching this! You are indeed right, the closest choice is deterministic while it shouldn't be.

I have one suggestion to implement it a bit cleaner.

File: mesa/space.py (hunk 1 above)
Comment (EwoutH) [inline mesa/space.py:464]:
```suggestion
                for p in agent.random.shuffle(pos):
```
I think with this single line change we could be done.

Comment (OrenBochman) [reply mesa/space.py:464]: Shuffling the list is a pretty simple solution so I don't mind doing that.

I printed all the pos lists for my simulation and most were pretty big but had a few closest locations. 

It would be faster to shuffle the shorter list like I did. However my idea needs to go over all the points which is slower.

But in terms of complexity the bulk of the work (complexity) is finding the smallest distance. Both ideas are the same. 

Before I found this method I was thinking of adding a distance representation masked layer ndarray so I could let numpy do this via a vectorized op. This would be cached by the agent after each move and reused.

But I did not explore this yet but it might be more efficient to build this for r =1 
check for a match, r=2 check.

Bottom line: I think to find a better answer would require bench-marking. It's probably not that important for this revision. Do as you see fit.

**Comment by OrenBochman** [comment]
Ok I see that the test passes and my example that uses this code is not showing the issue anymore.  SO +1

**Review by OrenBochman — COMMENTED** [review]
Ok I see that the test passes and my example that uses this code is not showing the issue anymore.  SO +1

**Comment by tpike3** [comment]
@EwoutH Do we want to wait for speed on this one or merge it and do a 2.3.1 to fix the bug?

**Review by EwoutH — APPROVED** [review]
Sorry that we did this die in red tape. I'm not sure if our benchmarks use this behavior, but for now correct behavior is more important than performance.

Thanks for the fix!
